v0/destinations/one_signal/transform.js

```diff
@@ -157,11 +157,11 @@ const trackResponseBuilder = (message, { Config }) => {
         properties[item.propertyName] ||
         properties[item.propertyName] === ""
       ) {
-        if (eventAsTags && typeof properties[item.propertyName] === "string") {
-          tags[`${event}_${[item.propertyName]}`] =
-            properties[item.propertyName];
-        } else if (typeof properties[item.propertyName] === "string") {
-          tags[item.propertyName] = properties[item.propertyName];
+        if (typeof properties[item.propertyName] === "string") {
+          const tagName = eventAsTags
+            ? `${event}_${[item.propertyName]}`
+            : item.propertyName;
+          tags[tagName] = properties[item.propertyName];
```

Comment:
@ItsSudip isn't `eventAsTags` is a boolean value? is it possible to use an object key as a boolean value?

Reply:
const tagName = eventAsTags  ? `${event}_${[item.propertyName]}` : item.propertyName; @shrouti1507 (ternary operator)

Reply:
Yeah, @koladilip. @shrouti1507 we are not using eventAsTags directly as some key of object we are using that as deciding variable.